aopy/preproc.py

```diff
@@ -619,6 +619,38 @@ def get_unique_trials(trial_idx, conditions, condition_name='target'):
                 corrected_trials = np.append(corrected_trials, trial)
     return corrected_trials
 
+def locate_trials_with_event(trial_events, event_codes, event_columnidx=None):
+    '''
+    Given an array of trial separated events, this function goes through and finds the event sequences corresponding to the trials
+    that include a given event. If an array of event codes are input, the function will find the trials corresponding to
+    each event code. 
+    Note: all events input through 'event_codes' must be located in the same column of 'trial_events'
+    
+    Args:
+        trial_events (ntr, nevents): Array of trial separated event codes
+        event_codes (int, str, list, or 1D array): Event code(s) to find trials for. Can be a list of strings or ints
+        event_column (int): Column index to look for events in. Keep as 'None' if all columns should be analyzed.
+        
```

Comment:
might be worth mentioning event_column indexing starts from 0.  The first column should be represented as event_column_idx  = 0